xapian-core/include/xapian/weight.h

```diff
@@ -1082,6 +1082,79 @@ class XAPIAN_VISIBILITY_DEFAULT PL2Weight : public Weight {
     double get_maxextra() const;
 };
 
+/// Xapian::Weight subclass implementing the PL2+ probabilistic formula.
+class XAPIAN_VISIBILITY_DEFAULT PL2PlusWeight : public PL2Weight {
```

Comment:
I think we may as well make protected members (below) private? Although, I see no harm in keeping them as protected at present unless in future a subclass derives from `PL2PlusWeight`. Same for `BM25PlusWeight` subclass.